var/lib/delphix-platform/ansible/10-delphix-platform/roles/delphix-platform/tasks/main.yml

```diff
@@ -302,3 +302,39 @@
     #
     - { key: 'AutoUpdate.Enabled', value: 'n' }
   when: platform == "azure"
+
+#
+# Customize the GCP linux environment.
+#
+# Update the override file for the GCP instance. This file gets
+# applied dynamically by running google_instance_setup script.
+#
+- blockinfile:
+    path: /etc/default/instance_configs.cfg.template
+    create: yes
+    block: |
+      #
+      # Disable the accounts daemon to prevent adding/removing
+      # users on the engine.
+      #
+      [Daemons]
+      accounts_daemon = false
+
+      #
+      # Disable user supplied startup/shutdown scripts from running on
+      # the engine.
+      #
+      [MetadataScripts]
+      shutdown = false
+      startup = false
+  when:
+    - platform == "gcp"
+  notify: "gcp config changed"
```

Comment:
will this run on the first boot of the system in GCP?

I worry the following will occur:
1. we'll configure this file during the build
2. we boot up the first time in GCP
3. this task runs, but doesn't modify the file (since it's the same as during the build)
4. since the file isn't changed, the handler doesn't run, so we don't run `google_instance_setup`

Reply:
If the file is the same then we don't need to run `google_instance_setup`.  We only need to run that if the file changes. I verified that after install the running file is updated:

```# cat /etc/default/instance_configs.cfg# This file is automatically created at boot time by the /usr/lib/python
# 3/dist-packages/google_compute_engine/instance_setup/instance_config.p
# y script. Do not edit this file directly. If you need to add items to
# this file, create or edit /etc/default/instance_configs.cfg.template
# instead and then run /usr/bin/google_instance_setup.

[Daemons]
accounts_daemon = false
clock_skew_daemon = true
ip_forwarding_daemon = true
network_daemon = true

[MetadataScripts]
shutdown = false
startup = false
```

Reply:
The idea of the notifier was to handle the upgrade case in the event that we do change this.

Reply:
:+1: OK, sounds good.